### PR TITLE
Fix backporting gaps to prevent old docs versions from showing up in search

### PR DESCRIPTION
**Pull request by LappleApple**
**What changed?**

This PR aims to close the backporting gap between v0.24.0 and v0.25.0 that is causing people to end up in the v0.24.0 version of the docs via searches.

**Why was this change made?**

So users don't get taken to old user docs pages.

**How was this change implemented?**

This updates v0.24.0 to include URLs, file names, sidebar references, and link updates made in v0.25.0.

**How did you validate the change?**

Ran yarn start a bunch of times until I got a clean result.

**Documentation Changes**

It's all about doc changes.

### CODE REVIEW

**Review by enekofb — COMMENTED** [review]
Lauri, 

what is the user flow where they end up  in v0.24:

- user goes to docs and click search, finds something -> click -> goes to v0.24 version -> docs not found or
- something else?

Btw, it seems there is some build issue see here https://github.com/weaveworks/weave-gitops/actions/runs/6112250021/job/16589310309

**Comment by LappleApple** [comment]
@enekofb The user flow is that a person will either come to the old pages through Google search or end up on these old pages after clicking inside our docs and unexpectedly being taken there. I'm aware of the build issues, I attributed it to this being a change affecting docs from several releases ago and was waiting till today to ask for insight/help in resolving.

**Comment by enekofb** [comment]
> I'm aware of the build issues, I attributed it to this being a change affecting docs from several releases ago and was waiting till today to ask for insight/help in resolving.

not sure if useful, but have raised an example of link fix [here]( https://github.com/weaveworks/weave-gitops/pull/3995)

in case you want to review how to view and fix them locally (in case you are not familiar) feel free to ping me

File: website/versioned_docs/version-0.24.0/enterprise/getting-started/aws-marketplace.mdx [hunk 1]
```diff
@@ -203,4 +202,4 @@ Your Weave GitOps installation is now ready!
 
 ## Next steps
 
-In our following [Get Started document](../getting-started/ui.mdx), we will walk you through logging into the GitOps Dashboard and deploying an application.
+In our Get Started document, we will walk you through logging into the GitOps Dashboard and deploying an application.
```
Comment (enekofb) [inline website/versioned_docs/version-0.24.0/enterprise/getting-started/aws-marketplace.mdx:205]: I guess that would be better to have a link to the getting started or to remove the section
Screenshot 2023-09-08 at 15 10 45

File: website/versioned_docs/version-0.24.0/enterprise/getting-started/install-enterprise.mdx [hunk 2]
```diff
@@ -421,7 +418,7 @@ gitops version
 
 ## Next steps
 
-In our following [Get Started document](../../getting-started/ui.mdx), we will walk you through logging into the GitOps Dashboard and deploying an application.
+In our following Get Started document, we walk you through logging into the GitOps Dashboard and deploying an application.
```
Comment (enekofb) [inline website/versioned_docs/version-0.24.0/enterprise/getting-started/install-enterprise.mdx:421]: same here, seems we are missing the getting started page

File: website/versioned_docs/version-0.24.0/gitops-templates/cli.mdx [hunk 3]
```diff
@@ -41,7 +41,7 @@ value of these two annotations:
 
 ## Installation
 
-See the Weave Gitops Enterprise [installation instructions](../installation/weave-gitops-enterprise/index.mdx#7-install-the-cli) for details on how to install the EE `gitops` CLI tool.
+See the Weave Gitops Enterprise [installation instructions](../enterprise/getting-started/install-enterprise.mdx#7-install-the-cli) for details on how to install the EE `gitops` CLI tool.
```
Comment (enekofb) [inline website/versioned_docs/version-0.24.0/gitops-templates/cli.mdx:44]:
```suggestion
See the Weave Gitops Enterprise [installation instructions](../enterprise/getting-started/install-enterprise.mdx/#install-the-weave-gitops-enterprise-cli-tool) for details on how to install the EE `gitops` CLI tool.
```

File: website/versioned_docs/version-0.24.0/gitops-templates/quickstart-templates.mdx [hunk 4]
```diff
@@ -9,7 +9,7 @@ import TierLabel from "../_components/TierLabel";
 # Quickstart GitOps Templates <TierLabel tiers="Enterprise" />
 
 `Quickstart` templates are [`GitOpsTemplate`s](https://docs.gitops.weave.works/docs/gitops-templates/templates/)
```
Comment (enekofb) [inline website/versioned_docs/version-0.24.0/gitops-templates/quickstart-templates.mdx:11]:
```suggestion
`Quickstart` templates are [`GitOpsTemplates`](https://docs.gitops.weave.works/docs/gitops-templates/templates/)
```

**Review by enekofb — COMMENTED** [review]
Some suggestions were added on the links. I definitely support not doing backporting changes going forward unless there in special occasions.

File: website/versioned_docs/version-0.24.0/enterprise/getting-started/aws-marketplace.mdx (hunk 1 above)
Comment (LappleApple) [reply website/versioned_docs/version-0.24.0/enterprise/getting-started/aws-marketplace.mdx:205]: @enekofb I didn't edit the docs content and don't see the value of updating these old versions.

Comment (LappleApple) [reply website/versioned_docs/version-0.24.0/enterprise/getting-started/aws-marketplace.mdx]: Removed section

File: website/versioned_docs/version-0.24.0/enterprise/getting-started/install-enterprise.mdx (hunk 2 above)
Comment (LappleApple) [reply website/versioned_docs/version-0.24.0/enterprise/getting-started/install-enterprise.mdx]: Removed section

**Review by enekofb — APPROVED** [review]

**Pull request merged**